renku/core/management/clone.py

```diff
@@ -53,29 +53,51 @@ def clone(
         os.environ["GIT_LFS_SKIP_SMUDGE"] = "1"
 
     try:
-        repo = Repo.clone_from(url, path, recursive=recursive, depth=depth, progress=progress)
-    except GitCommandError as e:
-        if not raise_git_except:
-            lines = progress.other_lines + progress.error_lines if progress else []
-            error = "".join([f"\n\t{line}" for line in lines if line.strip()])
-            message = f"Cannot clone remote Renku project: Git exited with code {e.status} and error message:\n {error}"
-            raise errors.GitError(message)
+        kwargs = {}
+
+        if checkout_rev:
+            kwargs["branch"] = checkout_rev
 
-        raise e
+        repo = Repo.clone_from(url, path, recursive=recursive, depth=depth, progress=progress, **kwargs)
```

Comment:
Because this is one of those kwargs that's just passed along to the actual `git clone` command. I believed if it was `branch=None` this would result in `git clone --branch git@github.com/org/repo` (so no branch specified) which would error, in which case this is cleaner than an If statement with the command repeated with/without branch.

But actually I just tested that hypothesis and it doesn't work that way, `branch=None` seems to be fine. So I'll change it to your suggestion :smile: